lib/srv/alpnproxy/proxy.go

```diff
@@ -101,6 +101,39 @@ func MatchByALPNPrefix(prefix string) MatchFunc {
 	}
 }
 
+// MatchMySQLConn return a pre-process function for MySQL connections that tries to extract MySQL server version
+// from incoming connection.
+func MatchMySQLConn(fn func(ctx context.Context, conn net.Conn) error) HandlerFuncWithInfo {
```

Comment:
The `MatchMySQLConn` is used a connection handler thus Match is a bit misleading. 
This function is use as middleware handler so can we name it based on the action that it does ? like `ExtractMySQLEngineVersion(...)`